src/main/groovy/au/org/ala/ecodata/converter/SciStarterConverter.groovy

```diff
@@ -64,88 +65,62 @@ class SciStarterConverter {
-                            } else {
-                                return null
+                            try {
+                                String url = props.url.replace("www.", "").replaceAll("\n","")
+
+                                new URL(url)
+                                url
+                            } catch (error) {
+                                log.info("${props.url} is not a valid URL, returning null...")
+                                null
                             }
-                        }],
+                        }
+                ],
+                'image'       : 'image',
                 'difficulty'  : [
                         'name'     : 'difficulty',
                         'transform': { props, target ->
-                            def difficulty
-                            if (props?.difficulty instanceof List) {
-                                difficulty = props.difficulty.findAll {
-                                    (it instanceof String) && it.capitalize() in ['Easy', 'Medium', 'Hard']
-                                }?.getAt(0)?.capitalize()
-                            }
-
-                            if (props?.difficulty instanceof String) {
-                                difficulty = props.difficulty.capitalize()
-                            }
-
-                            difficulty
+                            [
+                                    'easy': 'Easy',
+                                    'medium': 'Medium',
+                                    'difficult': 'Hard',
+                                    'unknown': null
+                            ][props.difficulty ? props.difficulty.label.toLowerCase() : 'unknown']
                         }
                 ],
-                'begin_date'  : [
+                'begin'  : [
                         'name'     : 'plannedStartDate',
                         'transform': { props, target ->
-                            SimpleDateFormat sdf = new SimpleDateFormat('yyyy-MM-dd');
-                            if (props.begin_date) {
-                                sdf.parse(props.begin_date)
-                            } else {
-                                new Date()
-                            }
+                            props.begin ? new Date(props.begin.longValue()) : new Date()
                         }],
-                'end_date'    : [
+                'end'    : [
                         'name'     : 'plannedEndDate',
                         'transform': { props, target ->
-                            SimpleDateFormat sdf = new SimpleDateFormat('yyyy-MM-dd');
-                            if (props.end_date) {
-                                sdf.parse(props.end_date)
-                            }
+                            props.end ? new Date(props.end.longValue()) : new Date()
```

Comment:
Without having seen the data this seems odd - are the dates coming as milliseconds since 1970?

Reply:
![image](https://github.com/user-attachments/assets/f90fe196-4a59-4768-ae0c-221526f292de)
Yep, the dates are milliseconds since 1970